doc/src/challenges/XXXX-rc.md

```diff
@@ -0,0 +1,130 @@
+# Challenge XXXX[^challenge_id]: Verify reference-counted Cell implementation
+
+- **Status:** Open
+- **Solution:** *Option field to point to the PR that solved this challenge.*
+- **Tracking Issue:** *Link to issue*
+- **Start date:** 2025/06/01
+- **End date:** 2025/12/31
+- **Reward:** *TBD*[^reward]
+
+-------------------
+
+
+## Goal
+
+The goal of this challenge is to verify Rc and its related Weak implementation. Rc is the library-provided building block that enables safe multiple ownership of data through reference counting for single-threaded cases, as opposed to the usual ownership types used by Rust. A related challenge verifies the Arc implementation, which is atomic multi-threaded.
```

Comment:
Please include source-code pointers as `Rc` is not as easy to search for. I'd appreciate if the same could be done for `Weak`and `Arc`.